libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/RouteUtils.kt

```diff
@@ -219,20 +219,15 @@ class RouteUtils {
     fun findCurrentBannerInstructions(
         currentStep: LegStep?,
         stepDistanceRemaining: Double
-    ): BannerInstructions? {
-        currentStep?.bannerInstructions()?.let {
+    ) = currentStep?.bannerInstructions()?.let {
             val instructions: List<BannerInstructions> = sortBannerInstructions(it)
-            for (instruction in instructions) {
-                val distanceAlongGeometry = instruction.distanceAlongGeometry().toInt()
-                if (distanceAlongGeometry >= stepDistanceRemaining.toInt()) {
-                    return instruction
-                }
+            instructions.firstOrNull { instruction ->
```

Comment:
Are you sure it is the equal with `for` looping by instructions? I mean your implementation changes the logic of this method. If this new logic is correct - super.... but if not ->... )

Reply:
Is this intentional @olegzil? It doesn't seem this is behaving as before. Overall implementation is different. Could you confirm?

Reply:
The logic has not changed. This is the sudo-code that describes the logic as it was implemented in the original code and remains the same in the modified code:
if currentStep?.bannerInstructions()? != null
*       generate a sorted BannerInstructions list
*       find a distanceAlongGeometry >= stepDistanceRemaining
*       if found return the banner instruction associated with the distance
*       else if the sorted instruction list is not empty return the first instruction
*       else return null
In short, find the banner instruction that contains a distance less then or equal to the step distance remaining. If one is found return that banner instruction. If not found, then if the banner list is null, return null, otherwise return the first instruction.
*       return null if the banner instruction list is null
*

Reply:
```kotlin
inline fun <T> Iterable<T>.firstOrNull(
    predicate: (T) -> Boolean
): T?
```

It's doing the same 👌 Look at the implementation 👀 

```kotlin
public inline fun <T> Iterable<T>.firstOrNull(predicate: (T) -> Boolean): T? {
    for (element in this) if (predicate(element)) return element
    return null
}
```

:shipit: @olegzil 
 
cc @Lebedinsky @RingerJK